include/heuristic/HeuristicMapper.hpp

```diff
@@ -294,7 +542,9 @@ class HeuristicMapper : public Mapper {
    * @param layer index of current circuit layer
    */
   void expandNode(const std::vector<std::uint16_t>& consideredQubits,
-                  Node& node, std::size_t layer);
+                  Node& node, std::size_t layer,
+                  const std::vector<std::uint16_t> singleQubitGateMultiplicity,
+                  const EdgeMultiplicity           twoQubitGateMultiplicity);
```

Comment:
Both of these could probably be const references. Please also check for all other functions taking these new parameters.